zerver/tests/test_realm_export.py

```diff
@@ -27,69 +21,78 @@ def test_export_as_not_admin(self) -> None:
+        mock_do_write_stats = Mock(spec=do_write_stats_file_for_realm_export,
+                                   return_value=FileNotFoundError)
+        with patch('zerver.lib.export.do_export_realm',
+                   side_effect=mock_do_write_stats()) as mock_export:
+            with self.assertRaises(FileNotFoundError):
+                DeferredWorker().consume(event)
             args = mock_export.call_args_list[0][1]
-            # TODO: Clean up the way we do the mocking here; we will
-            # want to mock do_export_realm in a way that captures its
-            # arguments but doesn't lead to (silent) error spam from
-            # do_write_stats_file_for_realm_export.
-            #
-            # Probably setting a `side_effect` makes sense?
-        self.assert_json_success(result)
-        self.assertEqual(args['realm'], self.admin.realm)
-        self.assertEqual(args['public_only'], True)
-        self.assertEqual(args['output_dir'].startswith('/tmp/zulip-export-'), True)
-        self.assertEqual(args['threads'], 6)
+            self.assertEqual(args['realm'], admin.realm)
+            self.assertEqual(args['public_only'], True)
+            self.assertIn('/tmp/zulip-export-', args['output_dir'])
+            self.assertEqual(args['threads'], 6)
 
     @override_settings(LOCAL_UPLOADS_DIR='/var/uploads')
     def test_endpoint_local_uploads(self) -> None:
-        with patch('zerver.lib.export.do_export_realm'):
-            with patch('zerver.views.public_export.queue_json_publish') as mock_publish:
-                result = self.client_post('/json/export/realm')
-                queue_data = mock_publish.call_args_list[0][0]
-                worker = mock_publish.call_args_list[0][0][0]
-        self.assert_json_success(result)
+        admin = self.example_user('iago')
+        self.login(admin.email)
 
+        with patch('zerver.views.public_export.queue_json_publish') as mock_publish:
+            result = self.client_post('/json/export/realm')
+            queue_data = mock_publish.call_args_list[0][0]
+            worker = mock_publish.call_args_list[0][0][0]
+        self.assert_json_success(result)
         mock_publish.assert_called_once()
         event = queue_data[1]
         self.assertEqual(worker, 'deferred_work')
         self.assertEqual(event['realm_id'], 1)
         self.assertEqual(event['user_profile_id'], 5)
         self.assertEqual(event['type'], 'realm_exported')
+        self.assertEqual(type(event['id']), int)
 
-        # Rest of test should match the previous test, but we're
-        # blocked on support for public export in LOCAL_UPLOADS_DIR
-        # backend.
+        mock_do_write_stats = Mock(spec=do_write_stats_file_for_realm_export,
+                                   return_value=FileNotFoundError)
+        with patch('zerver.lib.export.do_export_realm',
+                   side_effect=mock_do_write_stats()) as mock_export:
```

Comment:
You can just pass an exception class as `side_effect` without the extra `Mock(...` setup; see for example how we use it in `zerver/tests/test_queue_worker.py`